src/scores/categorical/contingency_impl.py

```diff
@@ -378,54 +381,124 @@ def threat_score(self) -> xr.DataArray:
         ts = cd["tp_count"] / (cd["tp_count"] + cd["fp_count"] + cd["fn_count"])
         return ts
 
-    def critical_success_index(self):
+    def critical_success_index(self) -> xr.DataArray:
         """
-        Often known as CSI.
-
         Identical to threat_score
-        Range: 0 to 1, 0 indicates no skill. Perfect score: 1.
 
-        https://www.cawcr.gov.au/projects/verification/#CSI
+        .. math::
+            \\text{threat score} = \\frac{\\text{true positives}}{\\text{true positives} + \\text{false positives} + \\text{true negatives}}
+
+        Returns:
+            An xarray object containing the critical success index
+
+        Notes:
+            - Range: 0 to 1, 0 indicates no skill. Perfect score: 1.
+            - Often known as CSI.
+            - "True positives" is the same as "hits"
+            - "False positives" is the same as "false alarms"
+            - "True negatives" is the same as "correct negatives"
+
+        References:
+            https://www.cawcr.gov.au/projects/verification/#CSI
         """
         return self.threat_score()
 
-    def peirce_skill_score(self):
+    def peirce_skill_score(self) -> xr.DataArray:
         """
-        Hanssen and Kuipers discriminant (true skill statistic, Peirce's skill score)
+        Identical to Hanssen and Kuipers discriminant and the true skill statistic
+
         How well did the forecast separate the "yes" events from the "no" events?
-        Range: -1 to 1, 0 indicates no skill. Perfect score: 1.
 
-        https://www.cawcr.gov.au/projects/verification/#HK
+        Returns:
+            An xarray object containing the Peirce Skill Score
+
+        .. math::
+            \\text{Peirce skill score} = \\frac{\\text{hits}}{\\text{hits} + \\text{misses}} - \\frac{\\text{false alarms}}{\\text{false alarms} + \\text{correct negatives}}
+
+        Notes:
+            - Range: -1 to 1, 0 indicates no skill. Perfect score: 1.
+            - "Hits" is the same as "true positives"
+            - "Misses" is the same as "false negatives"
+            - "False alarms" is the same as "false positives"
+            - "Correct negatives" is the same as "true negatives"
+
+        References:
+            https://www.cawcr.gov.au/projects/verification/#HK
```

Comment:
You can also add the original citation if you want 

Peirce, C.S., 1884. The numerical measure of the success of predictions. Science, (93), pp.453-454.

Reply:
If adding citation, here is the DOI: https://doi.org/10.1126/science.ns-4.93.453.b

Reply:
Although, the DOI isn't resolving for me. Not quite sure why? This is the URL that does resolve for me: https://www.science.org/doi/10.1126/science.ns-4.93.453.b

Reply:
I will resolve the citation in a separate issue, thanks.

Reply:
I created an issue for this citation